### PR TITLE
feat:Allow to opt out from trusted profile creation

### PR DESCRIPTION
**Pull request by aatreyee2506**
### Description

Added a variable to allow the users to opt out of trusted profile creation.[Git issue](https://github.ibm.com/GoldenEye/issues/issues/11391)

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Added a variable to allow the users to opt out of trusted profile creation.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by aatreyee2506** [comment]
cc : @SarikaSinha

File: main.tf [hunk 1]
```diff
@@ -67,7 +67,7 @@ locals {
 
 
 module "trusted_profile" {
-  count                       = (var.logs_agent_enabled && var.logs_agent_iam_mode == "TrustedProfile") ? 1 : 0
```
Comment (Aashiq-J) [inline main.tf:70]: Need to add the same logic to line number 110.

**Review by Aashiq-J — COMMENTED** [review]
Need to make the same change to the internal module as well.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I'm not sure this is the correct approach. We need to provide the ability for the user to pass an existing trusted profile ID. If the user does not pass one, the module should create one

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**